### ISSUE TITLE
Broken with Browserify

### ISSUE THREAD

**Issue by leeola** [showdownjs/showdown#50]
I attempted to use showdown with browserify and ran into a failure. The failure is due to Showdown trying to access the file system, based on the existence of require/etc.

My main question is.. why, why is it even doing this? If we look at the code in question..

```
//
// Automatic Extension Loading (node only):
//

if (typeof module !== 'undefind' && typeof exports !== 'undefined' && typeof require !== 'undefind') {
	var fs = require('fs');

	if (fs) {
		// Search extensions folder
		var extensions = fs.readdirSync((__dirname || '.')+'/extensions').filter(function(file){
			return ~file.indexOf('.js');
		}).map(function(file){
			return file.replace(/\.js$/, '');
		});
		// Load extensions into Showdown namespace
		Showdown.forEach(extensions, function(ext){
			var name = stdExtName(ext);
			Showdown.extensions[name] = require('./extensions/' + ext);
		});
	}
}
```

Why is `fs` even being used here? This seems like a complex solution for something that a simple `index.js` file would handle just fine. On top of that, if an `index.js` file was used then Browserify would work out of the box.

**Comment by tivie** [comment]
I'm currently looking into this. Showdown will have a major rewriting as keeping up with the perl version doesn't make much sense now. Since we'll probably move each
extension to its own repository, there will be no reason to call fs.

Regardless, the ideal way to approach the problem is abstracting the
library from the platform it's running on, in a similar fashion to
underscore.js method. This will make it compatible with node and browser
out of the box and, consequently, compatible with browserify. It will also
be easier to make the lib compatible with AMD and commonjs.

**Comment by zallek** [comment]
It's fixed in the showdown2 branch.
Note: You now need to instantiate converter using **camelcase** notation. 
`var converter = new Showdown.Converter();`

**Comment by tivie** [comment]
fixed in eae5f0e
